runtime/src/main/java/org/corfudb/runtime/DistributedCheckpointerHelper.java

```diff
@@ -19,27 +23,124 @@ public class DistributedCheckpointerHelper {
+            updateCompactionControlsTable(CompactorMetadataTables.INSTANT_TIGGER_WITH_TRIM, UpdateAction.PUT);
+        } else {
+            updateCompactionControlsTable(CompactorMetadataTables.INSTANT_TIGGER, UpdateAction.PUT);
+        }
+    }
+
+    /**
+     * Update the CompactionControlsTable with given Key and Action.
+     * Return if compaction is STARTED at the time of the update.
+     *
+     * @param stringKey the control operation in StringKey format
+     * @param action the action to perform
+     * @return if compaction is STARTED at the time of the update
+     */
+    private boolean updateCompactionControlsTable(StringKey stringKey, UpdateAction action) {
+        log.info("Updating CompactionControlsTable with Key:{}, Action:{}", stringKey, action);
+
+        boolean isCompactionInProgress = false;
+        Table<StringKey, RpcCommon.TokenMsg, Message> compactionControlsTable =
+                compactorMetadataTables.getCompactionControlsTable();
+
+        for (int retry = 1; retry <= CompactorMetadataTables.MAX_RETRIES; retry++) {
+            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                updateCompactionControlsTable(txn, compactionControlsTable, stringKey, action);
+
+                CheckpointingStatus managerStatus = (CheckpointingStatus) txn.getRecord(
+                        CompactorMetadataTables.COMPACTION_MANAGER_TABLE_NAME,
+                        CompactorMetadataTables.COMPACTION_MANAGER_KEY).getPayload();
+                isCompactionInProgress = managerStatus != null && managerStatus.getStatus().equals(StatusType.STARTED);
+                log.info("During updateCompactionControlsTable, isCompactionInProgress? {}", isCompactionInProgress);
+
+                txn.commit();
+                break;
+            } catch (TransactionAbortedException tae) {
+                if (tae.getAbortCause() == AbortCause.CONFLICT) {
+                    log.error("Another transaction has updated the CompactionControlsTable while " +
+                            "updating with Key:{}, Action:{}. Abort retrying.", stringKey, action);
+                    throw tae;
+                }
+            } catch (Exception e) {
+                if (retry == CompactorMetadataTables.MAX_RETRIES) {
+                    log.error("Failed to update CompactionControlsTable with Key:{}, Action:{} after retry.",
+                            stringKey, action, e);
+                    throw e;
+                }
+                log.warn("Unable to write Key:{}, Action:{} to CompactionControlsTable. Retrying for {} / {}.",
+                        stringKey, action, retry, CompactorMetadataTables.MAX_RETRIES);
+
+                try {
+                    TimeUnit.SECONDS.sleep(CompactorMetadataTables.TABLE_UPDATE_RETRY_SLEEP_SECONDS);
+                } catch (InterruptedException ie) {
+                    Thread.currentThread().interrupt();
+                    throw new IllegalStateException(ie);
```

Comment:
Optional comment: would it be better to throw this InterruptedException itself.

Reply:
I changed to just log a meesage without throwing the InterruptedException